druid/src/core.rs

```diff
@@ -962,8 +957,8 @@ impl BaseState {
         self.has_active |= child_state.has_active;
         self.has_focus |= child_state.has_focus;
         self.children_changed |= child_state.children_changed;
-        self.request_focus = self.request_focus.or(child_state.request_focus);
-        self.timers.extend(&child_state.timers);
+        self.request_focus = child_state.request_focus.take().or(self.request_focus);
+        self.timers.extend(&mut child_state.timers.drain());
```

Comment:
Was just looking at this yesterday; there's a small optimization we could make where if child has timers and we don't, we just do a `mem::swap`, which will save us an allocation?